function.go

```diff
@@ -0,0 +1,60 @@
+/*
+* Copyright 2020 New Relic Corporation. All rights reserved.
+* SPDX-License-Identifier: Apache-2.0
+ */
+
+package nriflex
+
+import (
+	"context"
+	"github.com/newrelic/nri-flex/internal/load"
+	"github.com/newrelic/nri-flex/internal/runtime"
+	"net/http"
+	"sync"
+)
+
+// nri-flex.main is not called by GCP, so everything has to happen here
+
+// Singleton, we only want to set this once
+var r runtime.Instance
+var log = load.Logrus
+
+// Target Function endpoint if you want to use HTTP  with the Cloud Scheduler
+func FlexHTTP(w http.ResponseWriter, req *http.Request) {
+	log.Debugf("FlexHTTP: enter")
+	run()
+	log.Debugf("FlexHTTP: exit")
+}
+
+type PubSubMessage struct {
+	Data []byte `json:"data"`
+}
+
+// Target Function endpoint if you want to use Pub/Sub  with the Cloud Scheduler
+func FlexPubSub(ctx context.Context, m PubSubMessage) error {
```

Comment:
@msummers-nr 
Nice Job ! 
One question, I might have missed it somewhere, just wondering how the PubSubMessage `m` data is passed to Flex for processing.

Reply:
It's not, I should add that as a comment. The Pub/Sub endpoint is used by the Cloud Scheduler to run Flex on a periodic basis- it's the Flex "heartbeat"

Reply:
@msummers-nr 
Got you!  Thanks for the explanation. 

Just a note, Flex could potentially process the message data from Pub/Sub and inject the data into NR. Currently, Flex running as a Lambda takes the Lambda event payload and processes it using `flex-lambda-ingest.yml` if it exists.   

Please see the Lambda implementation below: 
https://github.com/newrelic/nri-flex/blob/9a972059d0cb485bc1548a4d47e11d25ed02e838/internal/runtime/lambda.go#L76